twitchpopups.htm

```diff
@@ -149,9 +157,33 @@ <h1 id="popuptext"></h1>
         $("#popuptext").animate({"opacity":0, "margin-left":"50px"}, 700);
     }
     
+    function formatEmotes(message, emotes) {
+		  //parse the message for html and remove any tags
+      var newMessage = $($.parseHTML(message)).text().toUpperCase().split("");
+
+			//replace any twitch emotes in the message with img tags for those emotes
+      if(twitchEmotes){
+        for (var emoteIndex in emotes) {
+          var emote = emotes[emoteIndex];
+          for (var charIndexes in emote) {
+            var emoteIndexes = emote[charIndexes];
+            if (typeof emoteIndexes == "string") {
+              emoteIndexes = emoteIndexes.split("-");
+              emoteIndexes = [parseInt(emoteIndexes[0]), parseInt(emoteIndexes[1])];
+              for (var i = emoteIndexes[0]; i <= emoteIndexes[1]; ++i) {
+                newMessage[i] = "";
+              }
+              newMessage[emoteIndexes[0]] = "<img class=\"emoticon\" src=\"https://static-cdn.jtvnw.net/emoticons/v1/" + emoteIndex + "/3.0\">";

```

Comment:
If you use template literals you wont have to escape anything
```
 newMessage[emoteIndexes[0]] = `<img class="emoticon" src="https://static-cdn.jtvnw.net/emoticons/v1/${emoteIndex}/3.0"/>`;
```

Reply:
nice done by d9d19fe